src/main/scala/temple/generate/server/go/auth/GoAuthServiceMainGenerator.scala

```diff
@@ -40,8 +40,9 @@ object GoAuthServiceMainGenerator {
   private[auth] def generateRouter(): String =
     FileUtils.readResources("go/genFiles/auth/main/router.go.snippet").stripLineEnd
 
-  private[auth] def generateHandlers(): String =
-    FileUtils.readResources("go/genFiles/auth/main/handlers.go.snippet").stripLineEnd
+  private[auth] def generateHandlers(usesMetric: Boolean): String =
+    if (usesMetric) FileUtils.readResources("go/genFiles/auth/main/handlers-metric.go.snippet").stripLineEnd
+    else FileUtils.readResources("go/genFiles/auth/main/handlers-no-metric.go.snippet").stripLineEnd
```

Comment:
Beautiful

Reply:
oi

Reply:
an engineering decision considering the constraints has been made